### PR TITLE
add typedocs to cf and cf-wallet

### PR DESCRIPTION
**Pull request by patience-tema-baron**
Adds typedoc to the build script for cf.js and cf-wallet.js.

Currently the docs are themed to use markdown, that way they can be viewed [on github](https://github.com/counterfactual/monorepo/tree/519b3872b7c09803775e815b449c097923f667c1/packages/cf-wallet.js/docs). In order to view them on github, we obviously can't gitignore them, even though that might otherwise be preferable.

Alternatively, we could theme them as html (though we'll need to find someplace else to publish them to):

Screenshot from 2019-06-11 15-06-05

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -35,6 +35,8 @@
     "tslint-config-airbnb": "5.11.1",
     "tslint-config-prettier": "1.18.0",
     "tslint-plugin-prettier": "2.0.1",
+    "typedoc": "^0.14.2",
```
Comment (snario) [inline package.json]: Do we need this in the root `package.json`?

File: packages/cf-wallet.js/tsconfig.json [hunk 2]
```diff
@@ -1,6 +1,7 @@
 {
   "extends": "../../tsconfig.json",
   "compilerOptions": {
+    "sourceRoot": ".",
```
Comment (snario) [inline packages/cf-wallet.js/tsconfig.json]: Why is this added?

File: packages/cf-wallet.js/package.json [hunk 3]
```diff
@@ -15,11 +15,11 @@
   "repository": "https://github.com/counterfactual/monorepo/tree/master/packages/cf.js",
   "license": "MIT",
   "scripts": {
-    "build": "tsc -b . && rollup -c",
+    "build": "tsc -b . && rollup -c && yarn docs",
```
Comment (snario) [inline packages/cf-wallet.js/package.json]: Maybe have a separate `docgen` script?

File: package.json (hunk 1 above)
Comment (patience-tema-baron) [reply package.json]: Is the goal to eventually use typedocs throughout the project? If so, it makes sense here, right? If no, I could just move it into the relevant projects.

File: packages/cf-wallet.js/tsconfig.json (hunk 2 above)
Comment (patience-tema-baron) [reply packages/cf-wallet.js/tsconfig.json]: For parity w/ cf.js. I'd removed it from cf-wallet.js when I wrote it initially, since it was leading to a build error. This pr includes a fix for that build error, and so I could bring this line back. That being said, cf-wallet.js seems to work fine without it. Same with cf.js. Perhaps we could just remove it from both.

File: packages/cf-wallet.js/package.json (hunk 3 above)
Comment (patience-tema-baron) [reply packages/cf-wallet.js/package.json]: We currently have a `docs` script which works independently of the other build commands. I'm just calling it from here to ensure that the docs get regenerated whenever we make a change to these packages. I could remove it, but I'm worried about us failing to regenerate the docs if we have to do so manually.

File: package.json (hunk 1 above)
Comment (snario) [reply package.json]: Always makes sense to use it in the relevant projects

File: packages/cf-wallet.js/tsconfig.json (hunk 2 above)
Comment (snario) [reply packages/cf-wallet.js/tsconfig.json]: Why do we need it?

Comment (patience-tema-baron) [reply packages/cf-wallet.js/tsconfig.json]: As far as I can tell, we don't use it at present because we're not actually using the sourcemaps we generate, at least not within the context of the playground. That being said, even if we were using sourcemaps, it looks like it's just the tsconfig default value. I'm not sure why we added it in the first place, but it's probably safe to remove.

Comment (patience-tema-baron) [reply packages/cf-wallet.js/tsconfig.json]: Went ahead and removed this line just to unblock things.
